lib/fse.h

```diff
@@ -45,6 +45,32 @@ extern "C" {
 #include <stddef.h>    /* size_t, ptrdiff_t */
 
 
+/*-*****************************************
+*  FSE_PUBLIC_API : control library symbols visibility
+******************************************/
+#if defined(__GNUC__) && (__GNUC__ >= 4)
+#  define FSE_PUBLIC_API __attribute__ ((visibility ("default")))
```

Comment:
The issue lies in the gcc / clang environment, even though your patch if primarily intended for Visual environment.
With this version of source code, fse symbols are going to be exposed in `libzstd` irrespective of `FSE_DLL_EXPORT` when compiled with gcc or clang.

I'll handle this point directly.